doc/Requirements.md

```diff
@@ -0,0 +1,374 @@
+
+Two PF with the same parameters MUST have the same identifier for those parameters.
+(Why: Persistence of parameter identifier across PF instances with the same configuration.
+      To identify parameters independently of the host machine and PF instance)
+
+### Parameter tree
+A parameter SHOULD be structured in a tree. Each parameter being a distinct tree leaf.
+(Why: Tree is a simple data structure that can be easily represented and is enough to map
+      underlined layers.)
+
+Each node of the tree SHOULD have its own identifier with the same
+characteristics (type, independence, ...) than a parameter.
+(Why: To represent the tree without treating the leaf nodes specifically.)
+
+The identifier of each node of the tree SHOULD be a combination of its parents.
+More specifically, if the identifier is a string it should be formated the same
+way as a file system path. Eg the parameter named "/root/child1/4/parameter1".
+(Why: Usual syntax to address trees.)
+
+
+# Syncer
+
+The PF philosophy is to map the hardware characteristics to parameters.
+
+A syncer MUST be mapped to one or more parameters.
+(Why: The hardware minimal access may be bigger than one parameter.)
+
+One parameter MUST NOT be mapped to two or more syncer.
+(Why: Which syncer should be responsible to retrieve the initial parameter value
+      if they are multiple per parameter?)
+
+A syncer MUST support retrieving the mapped parameters value from the mapped hardware.
+(Why: to retrieve a parameter value at the start of the PF.)
+
+A syncer MUST support setting the mapped parameters value to the mapped hardware.
+(Why: to synchronise hardware on parameter change.)
+
+Syncers MUST retrieve and set the parameters value from the PF core.
+(Why: to be able to synchronise with hardware.)
+
+This API MAY be a packed parameter structure, following the C abi without padding.
+(Note: This is what the reference implementation does.)
+(Why: ???)
+
+## Introspection
+The syncer API SHOULD allow introspection of the mapped parameters.
+(Why: the parameter structure may be useful for the syncer to communicate with the hardware.)
+
+## Plugins
+The PF MUST be able to create syncers.
+(Why: to bind on the corresponding parameters.)
+
+### Definition
+The PF creates syncer using syncer builder.
+
+### Identifier
+All syncer mapping the same hardware SHOULD have their builder regrouped in a syncer library.
+(Why: to be able to link a group of parameters and a given hardware.)
+
+A syncer builder MUST have a unique identifier in its containing syncer library.
+(Why: To uniquely identify the syncer that should bind on parameters.
+      Given that the syncer library has already been specified.)
+
+A syncer library MUST have a unique identifier in the host system.
+(Why: To identify the library associated to parameters.)
+
+Syncer build and syncer library identifiers SHOULD be strings.
+(Why: The reference application does so.)
```

Comment:
I don't think this requirement is relevant. Also, rather than mentioning "unique identifiers", I'd rather state that a syncer library should be unambiguously retrievable (same for a syncer builder within a library). This suppresses the need for a an identifier to be a string and lets this be an implementation detail.